t/t7519-status-show-ignored-directory.sh

```diff
@@ -0,0 +1,117 @@
+#!/bin/sh
+#
+#
+
+test_description='git status collapse ignored'
+
+. ./test-lib.sh
+
+
+cat >.gitignore <<\EOF
+*.ign
+ignored_dir/
+!*.unignore
+EOF
+
+test_expect_success 'setup' '
+	mkdir -p tracked tracked_ignored \
+		ignored untracked \
+		untracked_ignored \
+		ignored_dir \
+		dir/ignored_dir \
+		dir/dir/tracked_ignored dir/dir/ignored \
+		dir/dir/untracked dir/dir/untracked_ignored \
+		dir/dir/ignored_dir \
+		dir/dir/ignored_with_sub_untracked \
+		dir/dir/ignored_with_sub_untracked/untracked \
+		dir/dir/untracked_with_sub_ignored \
+		dir/dir/untracked_with_sub_ignored/ignored &&
+
+	touch tracked/tracked \
+		ignored/ignored_1.ign \
+		ignored/ignored_2.ign tracked_ignored/tracked_1 \
+		tracked_ignored/tracked_2 \
+		tracked_ignored/ignored_1.ign \
+		tracked_ignored/ignored_2.ign \
+		untracked_ignored/ignored_1.ign \
+		untracked_ignored/ignored_2.ign \
+		ignored_dir/test \
+		ignored_dir/test.unignore \
```

Comment:
This test is confirming the behavior that you cannot unignore a file that is contained in an ignored directory. I can split it out into its own case, if desired...